docs/posts/lms-for-data/index.qmd

```diff
@@ -0,0 +1,690 @@
+---
+title: "Using language models for data"
+author: "Cody Peterson"
+date: "2024-02-05"
+image: thumbnail.png
+categories:
+    - blog
+    - llms
+    - duckdb
+---
+
+## Overview
+
+This post will give an overview of how (large) language models (LMs) fit into
+data engineering, data analyst, and data science workflows.
+
+## Use cases
+
+There are three main use cases for language models for data practitioners:
+
+1. Synthetic data generation
+2. Natural language processing
+3. Writing analytic code
+
+We'll describe each and then demonstrate them with code.
+
+## Setup
+
+We'll use [Marvin](https://askmarvin.ai), the AI engineering toolkit, alongside
+[Ibis](https://ibis-project.org), the data engineering toolkit, to demonstrate
+the capabilities of language models for data using the default DuckDB backend.
+
+:::{.callout-tip}
+We'll use a cloud service provider (OpenAI) to demonstrate these capabilities.
+In a follow-up post, we'll explore using local open source language models to
+achieve the same results.
+:::
+
+With Marvin and Ibis, you can replicate the workflows below using other AI
+service providers, local language models, and over 20+ data backends!
+
+You'll need to install Marvin and Ibis to follow along:
+
+```{.bash}
+pip install 'ibis-framework[duckdb,examples]' marvin
+```
+
+Then import them and turn on Ibis interactive mode:
+
+```{python}
+import ibis  # <1>
+import marvin  # <2>
+
+from pydantic import BaseModel, Field  # <3>
+
+ibis.options.interactive = True  # <4>
+ibis.options.repr.interactive.max_rows = 6  # <5>
+```
+
+```{python}
+# | code-fold: true
+# | echo: false
+_ = ibis.get_backend().raw_sql("set enable_progress_bar = false")
+```
+
+1. Import Ibis, the data engineering toolkit
+2. Import Marvin, the AI engineering toolkit
+3. Import Pydantic, used to define data models for Marvin
+4. Set Ibis to interactive mode to display the results of our queries
+5. Set the maximum number of rows to display in interactive mode
+
+## Synthetic data generation
```

Comment:
good use case. maybe point out that data is important for models to learn and it's often expensive/time-consuming to collect data. data synthesis solves this problem